docs/cli/app-types.md

```diff
@@ -0,0 +1,141 @@
+# .NET Core CLI types of portability of applications
```

Comment:
This title is a bit confusing to me. Maybe something like types of portable applications?